python/cudf/cudf/core/reshape.py

```diff
@@ -216,10 +229,17 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     if axis == 1:
 
         assert typs.issubset(allowed_typs)
+        if not typs.issubset(allowed_typs):
+            raise TypeError(
+                "Can only concatenate Series and DataFrame objects when axis=1"
+            )
         df = cudf.DataFrame()
         _normalize_series_and_dataframe(objs, axis=axis)
 
-        objs, match_index = _align_objs(objs)
+        if join == "inner":
+            objs, match_index = _align_objs(objs, how="inner")
+        else:
+            objs, match_index = _align_objs(objs)
```

Comment:
Since we only support `inner` and `outer` in both we can just explicitly pass it
```suggestion
        objs, match_index = _align_objs(objs, how=join)
```

Reply:
Makes sense!